### ISSUE TITLE
BUG: `floor_divide` returns strange values for integer division by zero

### ISSUE THREAD

**Issue by jakevdp** [google/jax#11990]
### Description

Example:
```python
In [1]: import numpy as np                                                                                                                                                                                                                                                                  

In [2]: np.floor_divide(2, 0)                                                                                                                                                                                                                                                               
<ipython-input-2-7ecbd88da223>:1: RuntimeWarning: divide by zero encountered in floor_divide
  np.floor_divide(2, 0)
Out[2]: 0

In [3]: import jax.numpy as jnp                                                                                                                                                                                                                                                             
jn
In [4]: jnp.floor_divide(2, 0)                                                                                                                                                                                                                                                              
Out[4]: DeviceArray(-2, dtype=int32, weak_type=True)
```

### What jax/jaxlib version are you using?

jax HEAD, jaxlib v0.3.14

### Which accelerator(s) are you using?

_No response_

### Additional System Info

_No response_

**Comment by hawkinsp** [comment]
I suspect this is working as intended: the behavior of division by zero will be implementation-defined. If you think it's important that we exactly match `numpy` we could add a `select` condition to `floor_divide`'s implementation.

**Comment by jakevdp** [comment]
Maybe, but "always return -2" is a strange convention

**Comment by mattjj** [comment]
Jake, could you add a priority label to this, and optionally reassign it to someone? (I can take it if you don't want to, where AIUI 'take it' means decide whether this WAI or else if we should add a `select` as Peter suggests, then add any implementation needed based on that decision.)

**Comment by jakevdp** [comment]
Done, thanks!

**Comment by mattjj** [comment]
I think this is working-as-intended.
